### ISSUE TITLE
[Feat] Connected components labelling algorithm

### ISSUE THREAD

**Issue by edgarriba** [kornia/kornia#866]
## 🚀 Feature

I propose to have the [Connected-component labelling](https://en.wikipedia.org/wiki/Connected-component_labeling) implemented in PyTorch so that can be used with the GPU and batched. Whether can be differentiable is open to discussion.

## Motivation

The connected components labeling is a well known technique that can be used across many computer vision applications.

Few existing implementations exist:
- Scipy - [`label`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.label.html)
- OpenCV - [`connectedComponents`](https://docs.opencv.org/4.5.1/d3/dc0/group__imgproc__shape.html#gaedef8c7340499ca391d459122e51bef5)
- Tensorflow - [`connected_components`](https://www.tensorflow.org/addons/api_docs/python/tfa/image/connected_components)

More resources:
 - https://www.youtube.com/watch?v=5AvHrIK-Kjc
 - https://gist.github.com/efirdc/5d8bd66859e574c683a504a4690ae8bc
 
## Pitch

I would expect to have the following functionality based on the tensorflow [implementation](https://github.com/tensorflow/addons/blob/v0.12.0/tensorflow_addons/image/connected_components.py#L27-L100).

`kornia.contrib.connected_components`.

```python
def connected_components(input: torch.Tensor) -> torch.Tensor:
    """Labels the connected components in a batch of images.

    Args:
        input (torch.Tensor): input tensor to be labelled in the shape (B, H, W).

    Return:
       torch.Tensor: the labelled tensor with shape (B, H, W).
    """
    output = /// IMPLEMENT ME
    return output
```

**Comment by mathfinder** [comment]
how is it going on？

**Comment by ducha-aiki** [comment]
Done in https://github.com/kornia/kornia/pull/1184
